akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/TestKitStorage.scala

```diff
@@ -47,15 +51,26 @@ sealed trait InMemStorage[K, R] extends InternalReprSupport[R] {
           Some(value.drop(fromInclusive).take(maxNum))
         else None)
 
+  def removeFirst(key: K): Unit = Try(eventsLog.get(key)).foreach(_.poll())
+
+  def first(key: K): Option[R] = Try(eventsLog.get(key)).toOption.flatMap { item =>
+    Try(item.element()).toOption.map(toRepr)
+  }
+
   def findOneByIndex(key: K, index: Int): Option[R] =
     Option(eventsMap.get(key))
       .flatMap {
         case (_, value) => if (value.size > index) Some(value(index)) else None
       }
       .map(toRepr)
 
-  def add(key: K, p: R): Unit =
+  def add(key: K, p: R): Unit = {
+    if (!Try(eventsLog.containsKey(key)).getOrElse(false)) {
```

Comment:
I don't think `containsKey` is expected to throw under normal conditions so no need for `Try`?
similar for `eventsLog.get` above.

Reply:
The signature it says `@throws NullPointerException if the specified key is null`. Do you think it would make sense to let an exception pop up in case?

Reply:
🤦 I misread that. It refers to the `key` as what I pass in and not what the hash man knows for the key aka. when the key is empty. Will change.